src/main/java/com/uber/cadence/common/RetryOptions.java

```diff
@@ -64,8 +64,11 @@ public static RetryOptions merge(MethodRetry r, RetryOptions o) {
     } else {
       builder.setBackoffCoefficient(DEFAULT_BACKOFF_COEFFICIENT);
     }
+    int maximumAttempts = merge(r.maximumAttempts(), o.getMaximumAttempts(), int.class);
+    if (maximumAttempts != 0) {
```

Comment:
no sure if I follow that. How is zero different from null in the processing? Could you provide the code link?

Reply:
@shijiesheng I think the point is setMaximumAttempts method throws exception if the value is 0.
The value 0 is default and a valid value if "expiration" is set.
Unfortunately I cannot build the project locally because of thrift 0.9.3 not available on linux currently but test like that should show it:
```java
public class RetryOptionsTest {
  @Test
  public void testUnlimitedRetryWithExpiration() throws NoSuchMethodException {
    RetryOptions topLevelOption =
        new RetryOptions.Builder().setExpiration(Duration.ofSeconds(5)).build();
    MethodRetry methodRetry =
        UnlimitedRetry.class.getMethod("retry").getAnnotation(MethodRetry.class);
    RetryOptions retryOptions = RetryOptions.merge(methodRetry, topLevelOption);
    Assert.assertEquals(
        "Top level options must take precedence", 5, retryOptions.getExpiration().getSeconds());
  }

  private interface UnlimitedRetry {
    @ActivityMethod
    @MethodRetry(expirationSeconds = 100)
    void retry();
  }
}
```